File: Example/javaapp/src/main/java/com/stripe/example/javaapp/fragment/PaymentFragment.java
package com.stripe.example.javaapp.fragment;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;

import com.stripe.example.javaapp.NavigationListener;
import com.stripe.example.javaapp.R;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.text.NumberFormat;
import java.util.Locale;

/**
 * The `PaymentFragment` allows the user to create a custom payment and ask the reader to handle it.
 */
public class PaymentFragment extends Fragment {

    @NotNull public static final String TAG = "com.stripe.example.fragment.PaymentFragment";

    @Nullable
    @Override
    public View onCreateView(
            @NotNull LayoutInflater layoutInflater,
            @Nullable ViewGroup container,
            @Nullable Bundle savedInstanceState
    ) {
        // Inflate the layout for this fragment
        final View view = layoutInflater.inflate(R.layout.fragment_payment, container, false);

        ((TextView) view.findViewById(R.id.amount_edit_text)).addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {}

            @Override
            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {}

            @Override
            public void afterTextChanged(Editable editable) {
                final String editableString = editable.toString();
                if (!editableString.isEmpty()) {
                    ((TextView) view.findViewById(R.id.charge_amount))
                            .setText(formatCentsToString(Integer.parseInt(editable.toString())));
                }
            }
        });

        view.findViewById(R.id.collect_payment_button).setOnClickListener(v -> {
            final FragmentActivity activity = getActivity();
            if (activity instanceof NavigationListener) {
                final String amount = ((TextView) view.findViewById(R.id.amount_edit_text))
                        .getText().toString();
                final String currency = ((TextView) view.findViewById(R.id.currency_edit_text))
                        .getText().toString();
                ((NavigationListener) activity).onRequestPayment(Integer.parseInt(amount), currency);
            }
        });

        view.findViewById(R.id.home_button).setOnClickListener(v -> {
            final FragmentActivity activity = getActivity();
            if (activity instanceof NavigationListener) {
                ((NavigationListener) activity).onRequestExitWorkflow();
            }
        });

        return view;
    }

    private String formatCentsToString(int cents) {
        return NumberFormat.getCurrencyInstance(Locale.US).format(cents / 100.0);
    }
}

File: Example/app/src/main/java/com/stripe/example/PaymentFragment.kt
package com.stripe.example

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_payment.view.*
import java.text.NumberFormat
import java.util.*

/**
 * The `PaymentFragment` allows the user to create a custom payment and ask the reader to handle it.
 */
class PaymentFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.fragment_payment, container, false)

        view.amount_edit_text.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
            }

            override fun afterTextChanged(editable: Editable?) {
                if (!editable.toString().isEmpty()) {
                    view.charge_amount.text = formatCentsToString(editable.toString().toInt())
                }
            }
        })

        view.collect_payment_button.setOnClickListener {
            if (activity is NavigationListener) {
                (activity as NavigationListener).onRequestPayment(
                        view.amount_edit_text.text.toString().toInt(),
                        view.currency_edit_text.text.toString())
            }
        }

        view.home_button.setOnClickListener {
            if (activity is NavigationListener) {
                (activity as NavigationListener).onRequestExitWorkflow()
            }
        }

        return view
    }

    private fun formatCentsToString(i: Int): String {
        return NumberFormat.getCurrencyInstance(Locale.US).format(i / 100.0)
    }
}
